packages/back-end/src/controllers/savedGroups.ts

```diff
@@ -0,0 +1,75 @@
+import { AuthRequest } from "../types/AuthRequest";
+import { Response } from "express";
+import { getOrgFromReq } from "../services/organizations";
+import {
+  createSavedGroup,
+  parseSaveGroupString,
+  updateSavedGroup,
+} from "../models/SavedGroupModel";
+
+// IMPORTANT: SavedGroups and Groups are very similar, but serve two different purposes. At the time of development 9/22 we are
+// quietly deprecating Groups. Initially groups were used with experiments to only include people in a group in an experiement.
+// SavedGroups are used with features flag rules where rules can say if "x is/is not in SavedGroup" do/don't show a feature
+
+export async function postSavedGroup(
+  req: AuthRequest<{
+    groupName: string;
+    owner: string;
+    attributeKey: string;
+    groupList: string;
+  }>,
+  res: Response
+) {
+  const { org } = getOrgFromReq(req);
+  const { groupName, owner, attributeKey, groupList } = req.body;
+
+  req.checkPermissions("createFeatures");
+
+  const values = parseSaveGroupString(groupList);
+
+  const savedGroup = await createSavedGroup({
+    values,
+    groupName,
+    owner,
+    attributeKey,
+    organization: org.id,
+  });
+
+  return res.status(200).json({
+    status: 200,
+    savedGroup,
+  });
+}
+
+export async function putSavedGroup(
+  req: AuthRequest<{
+    id: string;
+    groupName: string;
+    owner: string;
+    attributeKey: string;
+    groupList: string;
+  }>,
+  res: Response
+) {
+  const { org } = getOrgFromReq(req);
+  const { id, groupName, owner, groupList, attributeKey } = req.body;
+
+  if (!id) {
+    throw new Error("Must specify saved group id");
+  }
+
+  req.checkPermissions("createFeatures");
+
+  const values = parseSaveGroupString(groupList);
+
+  await updateSavedGroup(id, org.id, {
+    values,
+    groupName,
+    owner,
+    attributeKey,
+  });
```

Comment:
I would add audit logging here with `req.audit`.  In addition, if the list of values changed, I think we need to fire webhooks since it could change the feature definitions.

Reply:
Let's talk through the webhooks at our 1:1 today - I was under the impression that since we were saving everything by reference, the next time the feature rule was evaluated, it would just grab the updated values.